packages/search-in-workspace/src/browser/search-in-workspace-widget.tsx

```diff
@@ -427,30 +427,29 @@ export class SearchInWorkspaceWidget extends BaseWidget implements StatefulWidge
         const searchOnType = this.searchInWorkspacePreferences['search.searchOnType'];
         if (searchOnType) {
             const delay = this.searchInWorkspacePreferences['search.searchOnTypeDebouncePeriod'] || 0;
+
             window.clearTimeout(this._searchTimeout);
-            this._searchTimeout = window.setTimeout(() => this.doSearch(e), delay);
+            this._searchTimeout = window.setTimeout(() => {
+                if (e.target) {
+                    const searchValue = (e.target as HTMLInputElement).value;
+
+                    if (this.searchTerm === searchValue) {
+                        return;
+                    } else {
+                        this.searchTerm = searchValue;
+                        this.performSearch();
+                    }
+                }
+            }, delay);
         }
     };
 
-    protected readonly onKeyDownSearch = (e: React.KeyboardEvent) => {
+    protected readonly searchByEnterKey = (e: React.KeyboardEvent) => {
```

Comment:
`onKeyDownSearch` is also arguably better than `searchByEnterKey` because it's handed over as a `onKeyDown` callback.